llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -965,6 +965,40 @@ static void updateDefinedRegisters(MachineInstr &MI, LiveRegUnits &Units,
       Units.addReg(MOP.getReg());
 }
 
+/// Find the DBG_INSTR_REF instruction that references the \p InstrNum
+static std::optional<MachineInstr *> findDebugInstrRef(MachineBasicBlock *MBB,
+                                                       unsigned InstrNum) {
+
+  for (auto &MI : *MBB) {
+    if (MI.isDebugRef())
+      if (MI.getOperand(2).getInstrRefInstrIndex() == InstrNum)
+        return &MI;
+  }
+  return std::nullopt;
+}
+
+/// Set the correct debug-instr-number and the operand index in case of a merge.
```

Comment:
Is there a way to rephrase this comment to explain what the "correct" number+ref for a merge is?

Reply:
In fact, this function doesn't seem to do anything specific to merges, so not give it a generic name that describes the abstract operation that it does (finding an instrref for a number and setting the instrnum to that)?